static/src/javascripts/projects/common/modules/commercial/contributions-service.js

```diff
@@ -0,0 +1,210 @@
+    const page = config.get('page');
+
+    // note, there is a race condition so we want to fetch this as late as possible to give a change for the geo local storage value to be set
+    const countryCode = geolocationGetSync();
+
+    const tracking = {
+        ophanPageId: ophan.pageViewId,
+        ophanComponentId: 'ACQUISITIONS_EPIC',
+        platformId: 'GUARDIAN_WEB',
+        clientName: 'frontend',
+        referrerUrl:
+            window.location.origin + window.location.pathname,
+    };
+
+    const targeting = {
+        contentType: page.contentType,
+        sectionName: page.section,
+        shouldHideReaderRevenue: page.shouldHideReaderRevenue,
+        isMinuteArticle: config.hasTone('Minute'),
+        isPaidContent: page.isPaidContent,
+        isSensitive: page.isSensitive,
+        tags: buildKeywordTags(page),
+        // This test is already subjected to the 3 checks below, but
+        // we're passing these properties to the Contributions
+        // service for consistency with DCR.
+        showSupportMessaging: !shouldNotBeShownSupportMessaging(),
+        isRecurringContributor: isRecurringContributor(),
+        lastOneOffContributionDate:
+            getLastOneOffContributionDate() || undefined,
+        mvtId: getMvtValue(),
+        countryCode,
+    };
+
+    return {
+        tracking,
+        targeting,
+    };
+};
+
+const checkResponseOk = response => {
+    if (response.ok) {
+        return response;
+    }
+
+    throw new Error(
+        `Contributions fetch failed with response code: ${response.status}`
+    );
+};
+
+export const setEpic = (id: string) => {
+    const payload = buildPayload();
+    const products = ['CONTRIBUTION', 'MEMBERSHIP_SUPPORTER'];
+    const componentType = 'ACQUISITIONS_EPIC';
+    const viewEvent = makeEvent(id, 'view');
+
+    getBodyEnd(payload)
+        .then(checkResponseOk)
+        .then(response => response.json())
+         .then(json => {
+            if (json && json.data) {
+                const { html, css, js, meta } = json.data;
+                const trackingCampaignId = `epic_${meta.campaignId}`;
+
+                emitBeginEvent(trackingCampaignId);
+                setupClickHandling(meta.abTestName, meta.abTestVariant, componentType, meta.campaignCode, products);
+
+                renderEpic(html, css)
+                    .then(([el, shadowRoot]) => {
+                        executeJS(shadowRoot || el, js);
+                        submitOphanInsert(meta.abTestName, meta.abTestVariant, componentType, products, meta.campaignCode)
+                        setupOphanView(
+                            el,
+                            viewEvent,
+                            meta.abTestName,
+                            meta.abTestVariant,
+                            meta.campaignCode,
+                            trackingCampaignId,
+                            componentType,
+                            products,
+                        )})
```

Comment:
Is there anything here which allows us to distinguish between epic inserts/views which were rendered by the remote service vs local?

Reply:
Not directly, but we can use check the AB tests on the pageview table here (as it will include the RemoteEpicVariants test and variant info in it).